datasets/imagenet/imagenet.py

```diff
@@ -0,0 +1,148 @@
+import os
+import shutil
+
+import datasets
+
+
+_CITATION = """\
+@article{imagenet15russakovsky,
+    Author = {Olga Russakovsky and Jia Deng and Hao Su and Jonathan Krause and Sanjeev Satheesh and Sean Ma and Zhiheng Huang and Andrej Karpathy and Aditya Khosla and Michael Bernstein and Alexander C. Berg and Li Fei-Fei},
+    Title = { {ImageNet Large Scale Visual Recognition Challenge} },
+    Year = {2015},
+    journal   = {International Journal of Computer Vision (IJCV)},
+    doi = {10.1007/s11263-015-0816-y},
+    volume={115},
+    number={3},
+    pages={211-252}
+}
+"""
+
+_HOMEPAGE = "https://image-net.org/index.php"
+
+_DESCRIPTION = """\
+ILSVRC 2012, commonly known as 'ImageNet' is an image dataset organized according to the WordNet hierarchy. Each meaningful concept in WordNet, possibly described by multiple words or word phrases, is called a "synonym set" or "synset". There are more than 100,000 synsets in WordNet, majority of them are nouns (80,000+). In ImageNet, we aim to provide on average 1000 images to illustrate each synset. Images of each concept are quality-controlled and human-annotated. In its completion, we hope ImageNet will offer tens of millions of cleanly sorted images for most of the concepts in the WordNet hierarchy.
+"""
+
+_VAL_PREP_SCRIPT = "https://raw.githubusercontent.com/soumith/imagenetloader.torch/master/valprep.sh"
+_DEFAULT_CONFIG_NAME = "2012"
+_NUM_IMAGENET1K_CLASSES = 1000
+
+
+class ImagenetConfig(datasets.BuilderConfig):
+    def __init__(self, name=_DEFAULT_CONFIG_NAME, num_classes=_NUM_IMAGENET1K_CLASSES, **kwargs):
```

Comment:
There is a bug in datasets where if we pass `data_dir` to `load_dataset` it gets added to `config_kwargs` ultimately leading to a new config build as per https://github.com/huggingface/datasets/blob/cd3ce34ab1604118351e1978d26402de57188901/src/datasets/builder.py#L362. Is there a way to fix this so I can avoid having defaults and then also pop the `"version"` from the `kwargs`?

Reply:
But the problem is that it creates a new config with a new identifier each time, is that okay?